contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,163 @@
+#instructions to install googleapiclient can be found in this bot's readme file.
+from __future__ import print_function
+import base64
+import httplib2
+import logging
+import mimetypes
+import os
+import re
+
+from googleapiclient import discovery, errors
+from email.mime.text import MIMEText
+from oauth2client.file import Storage
+
+class GmailHandler(object):
+    '''
+    This plugin facilitates sending emails from a Gmail account
+    from within Zulip. It looks for messages starting with '@gmail'.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to send emails from a
+            personal gmail account. Users should preface
+            messages with "@gmail".
+
+            Enter messages in the following format:
+            '@gmail <recipient_email> "<subject>" "<body>" --label <label>'
+            The --label tag is optional and can be left out.
+            '''
+
+    def triage_message(self, message, client):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+        self.bot_email = client.email
+
+        # This next line of code is defensive
+        if message['display_recipient'] == 'gmail':
+            return False
+        return original_content.startswith('@gmail')
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        try:
+            recipient, subject, body, label = parse_input(original_content)
+        except ValueError:
+            recipient, subject, body = parse_input(original_content)
+            label = False
+        except TypeError:
+            raise Exception("The input format used was incorrect. The correct format is: "
```

Comment:
how stop the current execution of `gmail.py`? Because if it continues to execute `gmail.py` it will throw an error and crash a couple of lines later.